src/binder/statement/create_table.rs

```diff
@@ -45,57 +45,73 @@ impl Binder {
                     }
                 }
 
-                // record primary key names declared by "primary key" keywords
-                // if a sql like " create table t (a int, b int, c int, d int, primary key(a, b)); "
-                // then after the for loop, the extra_pk_name will contain "a" and "b"
-                let mut extra_pk_name: HashSet<String> = HashSet::new();
-                for constraint in constraints {
-                    match constraint {
-                        TableConstraint::Unique {
-                            is_primary,
-                            columns,
-                            ..
-                        } if *is_primary => columns.iter().for_each(|indent| {
-                            extra_pk_name.insert(indent.value.clone());
-                        }),
-
-                        _ => todo!(),
+                // column name -> index in `ordered_pk_ids` vector
+                let ordered_pk_pos = Binder::get_extra_pks(constraints);
+                // whether we have primary key declared by "primary key(c1, c2..) syntax"
+                let flag = !ordered_pk_pos.is_empty();
+                // ordered primary ids declared by "primary key(c1, c2..) syntax"
+                let mut ordered_pk_ids = vec![0; ordered_pk_pos.len()];
+                let mut col_catalogs = vec![];
+
+                for (idx, colum_def) in columns.iter().enumerate() {
+                    let mut is_primary = false;
+                    let mut col = ColumnCatalog::from(colum_def, &mut is_primary);
+                    if flag && is_primary {
+                        // we can't support sql query like
+                        // "create table (a int primary key, b int not null, primary key(b));"
+                        // declaring pk both in column's option and 'primary key(c1..)' syntax"
+                        return Err(BindError::NotSupportedTSQL);
+                    } else if is_primary {
+                        // primary key is declared only in column's options
+                        ordered_pk_ids.push(idx as ColumnId);
```

Comment:
So the default behavior of `create table (a int primary key, b int primary key)` will be a composite primary key of `(a, b)`? From my perspective, it's better to only allow one single pk in this case, so that users must specify whether it's `(a, b)` or `(b, a)` for composite pk?